packages/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift

```diff
@@ -39,8 +57,8 @@ extension ExpoSwiftUI {
      Initializes a SwiftUI hosting view with the given SwiftUI view type.
      */
     init(viewType: ContentView.Type, props: Props, appContext: AppContext) {
-      self.contentView = ContentView()
-      let rootView = AnyView(contentView.environmentObject(props).environmentObject(shadowNodeProxy))
+      self.contentView = ContentView(props: props)
+      let rootView = AnyView(contentView.environmentObject(shadowNodeProxy))
```

Comment:
passing props by `environmentObject` has performance impact because we don't need the props in the whole swiftui tree. this pr tries to pass by `init(props:)` instead.